.github/workflows/haskell-ci.yml

```diff
@@ -172,7 +172,7 @@ jobs:
           cabal-docspec --version
       - name: install hlint
         run: |
-          if [ $((HCNUMVER < 90000)) -ne 0 ] ; then HLINTVER=$(cd /tmp && (${CABAL} v2-install -v $ARG_COMPILER --dry-run hlint  --constraint='hlint >=3.3 && <3.4' |  perl -ne 'if (/\bhlint-(\d+(\.\d+)*)\b/) { print "$1"; last; }')); echo "HLint version $HLINTVER" ; fi
+          if [ $((HCNUMVER < 90000)) -ne 0 ] ; then HLINTVER=$(cd /tmp && (${CABAL} v2-install -v $ARG_COMPILER --dry-run hlint  --constraint='hlint >=3.5 && <3.6' |  perl -ne 'if (/\bhlint-(\d+(\.\d+)*)\b/) { print "$1"; last; }')); echo "HLint version $HLINTVER" ; fi
```

Comment:
Looks like this breaks GHC8 build - perhaps we need to change the condition so that this HLint version runs on GHC9?

I only wanted to run HLint on one GHC (otherwise versions get complicated), so I wrote this in `cabal.haskell-ci`:
```yaml
hlint-job: 8.10.7
```

Practically it's not important which GHC it runs on, just that it builds and runs on at least one, so feel free to change (or revert) to anything that makes sense. :slightly_smiling_face:

Reply:
OK, I'll try running it on 9.2.4.